Remove metric scrape interval

schmikei opened the pull request:
It makes little sense to have this functionality across the package and the invoker in main should just be the one that spins up the goroutine.

This also simplifies the shared config object.

schmikei committed:
remove metric scrape interval as it should be up to the invoker how to run

BinaryFissionGames commented on collector/collector.go:
```diff
@@ -167,21 +164,43 @@ func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
+		case <-ticker.C:
+			metricCh := make(chan prometheus.Metric, 5)
+			go func() {
+				scrapeResults := []prometheus.Metric{}
```
Should this be `e.scrapeResults`?

Otherwise, this doesn't seem to get used anywhere

BinaryFissionGames commented on collector/collector.go:
```diff
@@ -167,21 +164,43 @@ func (e *Exporter) Describe(ch chan<- *prometheus.Desc) {
-			ch <- scrapeResults[idx]
-		}
-	}
+	e.scrape(ch)
```
This makes it so that a scrape will always happen when the metrics endpoints is hit, even if ScrapeInterval is > 0.

Should this still be kept intact? 

I do think there should be a mux protecting e.scrapeResults, since there could be concurrent read/writes.

schmikei replied:
True let me take a second pass at this

vutkin committed:
feat: add gh workflows (#265)

dependabot[bot] committed:
Bump github.com/BurntSushi/toml from 1.2.0 to 1.2.1 (#271)

Bumps [github.com/BurntSushi/toml](https://github.com/BurntSushi/toml) from 1.2.0 to 1.2.1.
- [Release notes](https://github.com/BurntSushi/toml/releases)
- [Commits](https://github.com/BurntSushi/toml/compare/v1.2.0...v1.2.1)

---
updated-dependencies:
- dependency-name: github.com/BurntSushi/toml
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.7.3 to 0.8.2 (#272)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.7.3 to 0.8.2.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.7.3...v0.8.2)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.13.0 to 1.14.0 (#274)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.13.0 to 1.14.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.13.0...v1.14.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.37.0 to 0.39.0 (#273)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.37.0 to 0.39.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.37.0...v0.39.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

vutkin committed:
fix: Fixed github action runs against forked repos (#280)

* chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

fix: fix typo

fix: fix typo

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

fix: fix typo

* chore: savepoint

chore: savepoint

feat: test gh action pr-cleanup (#2)

* feat: test gh action pr-cleanup

* chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

* chore: savepoint

* chore: savepoint

* chore: savepoint

* chore: savepoint

* chore: savepoint

* chore: savepoint

* chore: savepoint

* feat: do a release 0.4.1 (#3)

* chore: savepoint

* chore: savepoint

* chore: savepoint

* feat: do a release 0.4.2 (#4)

* chore: savepoint

* chore: savepoint (#5)

* chore: savepoint

* chore: savepoint

* chore: savepoint

* feat: do release 0.4.4 (#6)

* feat: do a release 0.4.4

* chore: savepoint

* chore: savepoint

* chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

chore: savepoint

vutkin committed:
fix: fix GitHub release failed with status: 403 (#282)

pelov committed:
Fix 0.4.0 compilation problem (#277)

* Fix parsing kingpin flags according to exporter-toolkit v0.8.2

Fixes iamseth/oracledb_exporter#275

Signed-off-by: Nikolay Pelov <npelov@gmail.com>

* Update README file to reflect the changed options from exporter-toolkit.

* Update release version to 0.4.1

---------

Signed-off-by: Nikolay Pelov <npelov@gmail.com>

pelov committed:
Improve logging after the switch to github.com/go-kit/kit/log (#278)

Signed-off-by: Nikolay Pelov <npelov@gmail.com>

vutkin committed:
refact: refact PR/release github-actions  (#286)

* refact: refact PR github action a bit

* chore: savepoint

* chore: savepoint

caiwc committed:
Caiwc/multi target (#234)

* feat(): support scrape multi target

rebase branch && update Makefile version

* add multi-target explain to readme

* docs: correct dns example port of readme

---------

Co-authored-by: wccai <wccai@easyops.cn>
Co-authored-by: wccai <wccai@easyops>

Yannig committed:
README: add example of image hosted on github image registry

dependabot[bot] committed:
Bump golang.org/x/net from 0.4.0 to 0.7.0 (#290)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.4.0 to 0.7.0.
- [Release notes](https://github.com/golang/net/releases)
- [Commits](https://github.com/golang/net/compare/v0.4.0...v0.7.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.39.0 to 0.40.0 (#291)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.39.0 to 0.40.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.39.0...v0.40.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.40.0 to 0.41.0 (#292)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.40.0 to 0.41.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.40.0...v0.41.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.8.2 to 0.9.0 (#294)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.8.2 to 0.9.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.8.2...v0.9.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.9.0 to 0.9.1 (#295)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.9.0 to 0.9.1.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.9.0...v0.9.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.41.0 to 0.42.0 (#296)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.41.0 to 0.42.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.41.0...v0.42.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

markruler committed:
Fix broken link (#298)

Yannig committed:
:construction_worker: Update packages versions + clean up

* Update packages versions + clean up
* Use Oracle image to retrieve binaries
* Use FQN for docker image

Yannig committed:
:bug: Fix oraclelinux image build

Yannig committed:
:wrench: Use a parameter to set oracle client version

Marquizee committed:
remove percent and hyphen (#302)

Some oracledb tables return names with % and -

schmikei committed:
try to resolve scrape interval of 0 implications

schmikei committed:
Merge branch 'master' of github.com:iamseth/oracledb_exporter into oracledb-integration-scheduled-only-in-main

schmikei committed:
fix scrape interval nil check

schmikei committed:
fix synchronization
